diagnostics/inspection.py

```diff
@@ -24,46 +24,48 @@
 from . import utils
 
 
+sys.setrecursionlimit(5000)
+
+
 def inspect_samples(y0: Tensor,
                     ts: Vector,
                     dt: Scalar,
                     sde: BaseSDE,
                     bm: BaseBrownian,
                     img_dir: str,
                     methods: Tuple[str, ...],
-                    options: Optional[Tuple] = None,
+                    options: Tuple,
+                    labels: Tuple[str, ...],
                     vis_dim=0,
-                    dt_true: Optional[float] = 2 ** -14,
-                    labels: Optional[Tuple[str, ...]] = None):
+                    dt_true: Optional[float] = 2 ** -14):
     if options is None:
         options = (None,) * len(methods)
     if labels is None:
         labels = methods
 
-    sde = copy.deepcopy(sde).requires_grad_(False)
-
-    solns = [
-        sdeint(sde, y0, ts, bm, method=method, dt=dt, options=options_)
-        for method, options_ in zip(methods, options)
-    ]
+    with torch.no_grad():
```

Comment:
Since this whole function doesn't require grads to be computed, why not just decorate the whole func? E.g. 
```
@torch.no_grad()
def inspect_samples...
```